src/main.rs

```diff
@@ -143,7 +144,42 @@ fn get_issue_tree_init_hashes(repo: &Repository, _: &clap::ArgMatches) -> i32 {
 
 // Porcelain subcommand implementations
 
-// ...
+/// new subcommand implementation
+///
+fn new_impl(repo: &Repository, matches: &clap::ArgMatches) -> i32 {
+    let sig = try_or_1!(repo.signature());
+
+    // get the message, either from the command line argument or an editor
+    let message = if let Some(paragraphs) = matches.values_of("message") {
+        // the message was supplied via the command line
+        paragraphs.map(str::to_owned).map(|p| (p + "\n").to_owned()) // paragraphs
+                  .chain(try_or_1!(repo.prepare_trailers(matches)).into_iter().map(|t| t.to_string()))
```

Comment:
@matthiasbeyer before you ask: you cannot simply write `.map(Trailer::to_string)` because `ToString::to_string()` takes a `&self`.